assets/components/stripeInlineForm/stripeInlineForm.jsx

```diff
@@ -136,14 +140,32 @@ function checkoutForm(props: {
     }
   };
 
+  const submitPaymentBaseClass = 'component-stripe-inline-form__submit-payment';
+  const cardElementBaseClass = 'component-stripe-inline-form__card-element';
+  const className = baseClass =>
+    (props.disable ? classNameWithModifiers(baseClass, ['disable']) : baseClass);
+
+
+  const formPointerEvents: string = props.disable ? 'pointer-events: none' : 'pointer-events: all';
+
   return (
-    <form className="component-stripe-inline-form" onSubmit={handleSubmit}>
+    <form className="component-stripe-inline-form" onSubmit={handleSubmit} style={formPointerEvents}>
       <label>
         <span className="component-stripe-inline-form__label-content">Enter credit/debit card details</span>
-        <CardElement className="component-stripe-inline-form__card-element" hidePostalCode style={stripeElementsStyle} />
+        <CardElement
+          className={className(cardElementBaseClass)}
+          hidePostalCode
+          style={stripeElementsStyle}
+        />
       </label>
       <ErrorMessage message={props.errorMessage} />
-      <button className="component-stripe-inline-form__submit-payment">Confirm card payment <SvgArrowRightStraight /></button>
+      <button
+        className={className(submitPaymentBaseClass)}
+        disabled={props.disable}
+      >
+        Confirm card payment
+        <SvgArrowRightStraight />
+      </button>
```

Comment:
Could we avoid inline styles and the extra logic needed to apply individual modifier classes by applying a modifier on the enclosing component?

```jsx
<form
  className={classNameWithModifiers('component-stripe-inline-form', props.disable ? ['disabled'] : [])}
  onSubmit={handleSubmit}
>
  ...
</form>
```

```scss
.component-stripe-inline-form--disabled {
  pointer-events: none;

  .component-stripe-inline-form__submit-payment {
    // Disabled styles.
  }

  .component-stripe-inline-form__submit-payment {
    // Disabled styles.
  }

}
```

Reply:
👍